apps/glusterfs/healthz.go

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright (c) 2019 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+
+	"github.com/boltdb/bolt"
+	"github.com/lpabon/godbc"
+)
+
+var (
+	clusterHealth = "ClusterHealth"
+)
+
+type dbchecker struct {
+	CurrentTime time.Time
+}
+
+func (d *dbchecker) Marshal() ([]byte, error) {
+	var buffer bytes.Buffer
+	enc := gob.NewEncoder(&buffer)
+	err := enc.Encode(*d)
+
+	return buffer.Bytes(), err
+}
+
+func (d *dbchecker) Unmarshal(buffer []byte) error {
+	dec := gob.NewDecoder(bytes.NewReader(buffer))
+	return dec.Decode(d)
+}
+
+func (*dbchecker) BucketName() string {
+	return BOLTDB_BUCKET_DBHEALTHZ
```

Comment:
To be honest I'm really not sure we want a special bucket just for tihs, especially since we're saving a single value. Wouldn't it be simpler and less code to simply set/get an new key under dbattributes?

@raghavendra-talur : I'm curious if you have an opinion about this. The current approach is functional but feels a little like overkill to me. Could I get a 2nd opinion from you?

Reply:
@raghavendra-talur let me  know your thoughts on this